data_steward/cdr_cleaner/cleaning_rules/README.md

```diff
@@ -0,0 +1,89 @@
+# Cleaning Rules
+All cleaning rules should be implemented as classes derived from `BaseCleaningRule`. The main objective of a cleaning rule is to generate queries returned by `get_query_specs()`. To do this a cleaning rule should use parameters `project_id`, `dataset_id` provided at runtime (e.g. when running `clean_cdr.py`).
+
+Some initial setup such as loading lookup tables may be needed for a cleaning rule to generate working queries; such setup should be done in `setup_rule()`. Cleaning rules should store lookups, intermediate data, and backup data in the location indicated by the `sandbox_dataset_id` parameter.
+
+## Implementation guidance
+ 
+- Cleaning rules should ideally be [idempotent](https://en.wikipedia.org/wiki/Idempotence)
+- Cleaning rules execute independently from one another, but may have implicit data interdependencies- be mindful of the order in which they run in the cleaning process.
+- If you are authoring a cleaning rule which requires some parameter other than the [universal parameters](#universal-parameters), check for a suitable [variable parameter](#variable-parameters) before creating a custom one specific for your cleaning rule and ensure that is documented in the same way as other parameters.
+
+# Cleaning rule configuration
+Cleaning rules depend on configuration parameters which must be supplied at runtime when cleaning a dataset. Some parameters are needed by all cleaning rules whereas others may only pertain to a few. These dependencies become difficult to manage without some overarching design. This section describes the design objective and summarizes existing configuration dependencies.
+
+## Parameters correctly supplied
+> Note: for these examples assume that cleaning fitbit only requires the additional parameters `mapping_dataset_id` and `pid_rid_map_tablename`
```

Comment:
The `pid_rid_map_tablename` table is assumed to be `_deid_map` in the `mapping_dataset_id` currently and as such a separate argument named `pid_rid_map_tablename` is not used.

It might be important to note that `RemoveFitbitDataIfMaxAgeExceeded` fitbit CR is using the `combined_dataset_id` because of its dependence on the person table

Reply:
I would debate that the reason it is parameterized is because the name is subject to change and shouldn't be hard coded or assumed.

Reply:
We recently determined that the table name was not subject to change- `"_deid_map"` had to be supplied in order to find the appropriate schema file.